### ISSUE TITLE
OOP not working?

### ISSUE THREAD

**Issue by mirceaciu** [brody2consult/cordova-sqlite-storage#496]
I am trying to wrap Sqlite's functionalities like so:

```
db = new function() {

this.dbOpen = null;

this.openDb = function (dbName, dbLocation) {
    this.dbOpen = window.sqlitePlugin.openDatabase({name: dbName, location: dbLocation});
};

this.callSuccess= function(){
    console.log('some success');
};

this.callFail= function(){
    console.log('we fail')
};

this.createTable = function(){

   var tx = this.dbOpen.beginTransaction();
    tx.executeSql("DROP TABLE IF EXISTS user");
    tx.executeSql("CREATE TABLE IF NOT EXISTS user (id INTEGER PRIMARY KEY, userName TEXT, userToken TEXT)");
    tx.end(function() {
        console.log('Optional success callback');
    }, function(e) {
        console.log("Optional error callback with message: " + e.message);
    });
};
}
```

Then inside onDeviceReady() I call my functions like so:

```
db.openDb("eatDatabase", "default");

db.createTable();
```

The openDb() will create the new database and return the success callback. But in the createTable() section it will generate an error

> Uncaught TypeError: undefined is not a function 

at

```
 var tx = this.dbOpen.beginTransaction();
```

Is something preventing me from using "this." ?

I am running on Apache Cordova, building and testing with an Android device and [sqlite-enterprise-free](https://github.com/litehelpers/Cordova-sqlite-enterprise-free/blob/evfree-ext-rc/www/SQLitePlugin.js)

**Comment by brody2consult** [comment]
I don't think `this` will work in your function for "this.createTable". Most of your code seems to wrapped in "db = new function() {" though I do not see the ending "}" so you should be able to add a "var self = this" line and use self in the "this.createTable" function.

Unfortunately I will be busy for the next 2-3 weeks with some urgent work. I recommend that you read up some more about how JavaScript functions and objects work. There is plenty of excellent material available for free on the internet. You may also want to check out how CoffeeScript works and generates JavaScript.

**Comment by mirceaciu** [comment]
You are right, I ommited the closing "}", I will edit the question to contain that. The ideea of having a wrapping function is to have acces to functions like "db.create" ,"db.createTable" and others like "user.insert", " user.delete" with parametres for all of this functions. I hope i'll find something that will make this oop concept work and that you will sort work out. Thank you!
